clang/include/clang/Basic/Attr.td

```diff
@@ -1497,6 +1497,30 @@ def IntelNamedSubGroupSize : InheritableAttr {
 def :
   MutualExclusions<[IntelReqdSubGroupSize, IntelNamedSubGroupSize, SYCLSimd]>;
 
+def SYCLIntelFPGAIpInterfaceCsr: InheritableAttr {
+  let Spellings = [CXX11<"intel", "ip_interface_csr">];
+  let LangOpts = [SYCLIsDevice, SilentlyIgnoreSYCLIsHost];
+  let Subjects = SubjectList<[Function], ErrorDiag>;
+  let Documentation = [SYCLIntelFPGAIpInterfaceCsrDocs];
+  let SupportsNonconformingLambdaSyntax = 1;
```

Comment:
We need to stop adding attributes that use non-conforming lambda syntax -- this is not valid C++ code. Is this strictly required? I'm guessing "yes" because it applies to kernels, but this pattern of "well, we already added it one place so now we need to do it everywhere" is really a concern.